nisystemlink/clients/auth/_auth_client.py

```diff
@@ -0,0 +1,33 @@
+"""Implementation of AuthClient."""
+
+from typing import Optional
+
+from nisystemlink.clients import core
+from nisystemlink.clients.core._uplink._base_client import BaseClient
+from nisystemlink.clients.core._uplink._methods import get
+
+from . import models
+
+
+class AuthClient(BaseClient):
+    def __init__(self, configuration: Optional[core.HttpConfiguration] = None):
+        """Initialize an instance.
+
+        Args:
+            configuration: Defines the web server to connect to and information about
+                how to connect. If not provided, an instance of
+                :class:`JupyterHttpConfiguration <nisystemlink.clients.core.JupyterHttpConfiguration>`
+                is used.
+
+        Raises:
+            ApiException: if unable to communicate with the Auth Service.
+        """
+        if configuration is None:
+            configuration = core.JupyterHttpConfiguration()
+
+        super().__init__(configuration, base_path="/niauth/v1/")
+
+    @get("auth")
+    def get_auth(self) -> models.AuthResponse:
```

Comment:
Something else I thought of: the only auth functionality we need for data spaces is to get workspaces by name, right? While this method could be useful, you could just call `/niuser/v1/workspaces` instead, which would have a lower testing burden.

Reply:
It may not fit the purposes of DataSpaces as per another internal discussion.

Reply:
I talked to @spanglerco and we agree that the `/auth` route is the correct way to get available workspaces.

Unrelated, I think I would prefer this method to be called `authenticate()` rather than `get_auth()`.